code/physics/physics.cpp

```diff
@@ -755,33 +763,48 @@ void physics_apply_whack(vec3d *impulse, vec3d *pos, physics_info *pi, matrix *o
 	vm_vec_rotate ( &local_torque, &torque, orient );
 
 	vec3d delta_rotvel;
-	vm_vec_rotate( &delta_rotvel, &local_torque, &pi->I_body_inv );
-	vm_vec_scale ( &delta_rotvel, (float) ROTVEL_WHACK_CONST );
-	vm_vec_add2( &pi->rotvel, &delta_rotvel );
+	vm_vec_rotate(&delta_rotvel, &local_torque, inv_moi);
+
+	// Goober5000 - pi->mass should probably be just mass, as specified in the header
+	vec3d delta_vel = *impulse * (1.0f / mass);
+
+	physics_apply_whack_direct(vm_vec_mag(impulse), pi, &delta_rotvel, &delta_vel, orient);
+}
+
+
+// This function applies the calculated delta rotational and linear velocities usually called by ^^^^ apply_whack, 
+// but also called directly by dock_whack_docked_object in objectdock.cpp which has to do some extra legwork
+// and doesn't set this function up using physics_apply_whack
```

Comment:
What's the difference between "direct" and non-direct?  What does "direct" mean in this case?

Reply:
Naming of this was a little bit of an issue. "Direct" is actually applying the velocities calculated by `physics_apply_whack' and possibly also `dock_whack_docked_object`, those two are where the actual physical calculations are done. I'd be open to a better naming scheme. `physics_calculate_whack` and `physics_apply_whack`?

Reply:
Yes, that sounds better.